dev/NavigationView/NavigationView_ApiTests/NavigationViewTests.cs

```diff
@@ -872,5 +872,26 @@ public void VerifyOverflowButtonToolTip()
                 Verify.IsTrue(testCondition, "ToolTip text should have been \"More\".");
             });
         }
+
+        [TestMethod]
+        public void VerifyHierarchicalNavigationTopModeMenuItemsSourceDoesNotCrash()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                var navView = new NavigationView();
+                Content = navView;
+
+                navView.PaneDisplayMode = NavigationViewPaneDisplayMode.Top;
+
+                var childItem = new NavigationViewItem() { Content = "Item 1.1" };
+                var parentItem = new NavigationViewItem() { Content = "Item 1", MenuItemsSource = new ObservableCollection<NavigationViewItem>() { childItem } };
+                navView.MenuItemsSource = new ObservableCollection<NavigationViewItem>() { parentItem };             
+
+                Content.UpdateLayout();
+
+                // If we don't get here, app has crashed. This verify signals the NavigationView was launched in Top mode without crashing.
+                Verify.IsTrue(true);
```

Comment:
Technically, this verify does not ensure we reached the end of the method. TAEF/MSTest don't really know how many verifies they should have run. If the test fails, we only know that because the test didn't report a result.

Reply:
Yes, I copied existing code here to verify "code did not cuase a crash" and since those were added to WinUI I figured it's fine using the same concept here.